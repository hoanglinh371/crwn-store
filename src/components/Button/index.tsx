import React from 'react';
import classNames from 'classnames/bind';

import styles from './Button.module.scss';

interface Props {
  children: React.ReactNode;
  type: 'button' | 'submit' | 'reset' | undefined;
  google?: boolean;
  inverted?: boolean;
  onClick?: () => void;
}

const cx = classNames.bind(styles);

const Button: React.FC<Props> = (props) => {
  const { children, type, google, inverted, onClick } = props;

  return (
    <button
      type={type}
      className={cx('container', { google, inverted })}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default Button;
