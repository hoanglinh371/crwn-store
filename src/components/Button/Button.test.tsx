import { cleanup, render, screen } from '@testing-library/react';

import Button from '.';

describe('Button component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Button type='button' google>
        Click here!
      </Button>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should_HaveType', () => {
    render(
      <Button type='button' google>
        Click here!
      </Button>,
    );
    expect(screen.getByLabelText('button')).toHaveAttribute('type', 'button');
  });

  it('Should_HaveContent', () => {
    render(
      <Button type='button' google>
        Click here!
      </Button>,
    );
    expect(screen.getByLabelText('button')).toHaveTextContent('Click here!');
    console.log(screen.getByLabelText('button'));
  });

  it('Should_HaveAllClassnames', () => {
    render(
      <Button type='button' google>
        Click here!
      </Button>,
    );
    expect(screen.getByLabelText('button')).toHaveClass('container', 'google');
  });
});
