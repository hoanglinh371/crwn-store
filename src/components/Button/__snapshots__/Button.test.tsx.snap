// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should_MatchSnapshot 1`] = `
<div>
  <button
    aria-label="button"
    class="container google"
    type="button"
  >
    Click here!
  </button>
</div>
`;
